Lib/test/test_gdb.py

```diff
@@ -889,6 +889,30 @@ def test_printing_builtin(self):
         self.assertMultilineMatches(bt,
                                     r".*\nbuiltin 'len' = <built-in method len of module object at remote 0x-?[0-9a-f]+>\n.*")
 
+    @unittest.skipIf(python_is_optimized(),
+                     "Python was compiled with optimizations")
+    def test_complex_expr_print(self):
+        tested_code = '''
+class Player:
+    def __init__(self):
+        self.name = 'Martin'
+        self.team = 'HC Sparta Praha'
+        self.skills = {'defence': 2}
+
+a = [0, 1, 2, Player()]
+b = 0
+id(0)'''
+
+        gdb_output = self.get_stack_trace(tested_code,
+                                          cmds_after_breakpoint=['py-up', 'py-print a[0]'])
+        self.assertRegex(gdb_output, r".*\nglobal 'a\[0\]' = 0\n.*")
+        gdb_output = self.get_stack_trace(tested_code,
+                                          cmds_after_breakpoint=['py-up', 'py-print a[3].name'])
+        self.assertRegex(gdb_output, r".*\nglobal 'a\[3\]\.name' = 'Martin'\n.*")
+        gdb_output = self.get_stack_trace(tested_code,
+                                          cmds_after_breakpoint=['py-up', 'py-print a[3].skills[defence]'])
```

Comment:
The  `skills[defence]` syntax seems weird to me; shouldn't this be `skills["defence"]` or `skills['defence']`?